internal/flypg/barman_restore.go

```diff
@@ -0,0 +1,301 @@
+package flypg
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/fly-apps/postgres-flex/internal/supervisor"
+)
+
+type BarmanRestore struct {
+	*Barman
+
+	recoveryTarget          string
+	recoveryTargetName      string
+	recoveryTargetTime      string
+	recoveryTargetAction    string
+	recoveryTargetInclusive string
+}
+
+const (
+	defaultRestoreDir = "/data/postgresql"
+	// ISO8601 is the format required for the barman restore targets.
+	ISO8601 = "2006-01-02T15:04:05-07:00"
```

Comment:
can we use `time.RFC3339` as the layout so that I can pass `Z` as the time zone? e.g. `2024-02-03T04:05:06Z`

Reply:
Postgres doesn't support it, unfortunately.